src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -1066,6 +1068,33 @@ private static void VerifyModelForOutVarInNotExecutableCode(SemanticModel model,
             Assert.Null(model.GetDeclaredSymbol(decl));
         }
 
+        private static void AssertTypeFromOperation(SemanticModel model, TypeSymbol expectedType, DeclarationExpressionSyntax decl)
+        {
+            var variableDeclarator = decl.Ancestors().OfType<VariableDeclaratorSyntax>().FirstOrDefault();
+            if (variableDeclarator?.ArgumentList?.FullSpan.Contains(decl.Span) == true)
+            {
+                // node in argument list is error case which operation doesn't support
+                return;
+            }
+
+            var foreachLoop = decl.Ancestors().OfType<ForEachVariableStatementSyntax>().FirstOrDefault();
+            if (foreachLoop?.Variable?.FullSpan.Contains(decl.Span) == true &&
+                foreachLoop?.Variable.IsKind(SyntaxKind.InvocationExpression) == true)
+            {
+                // invalid syntax case where operation is not supported
+                return;
+            }
+
+            var typeofExpression = decl.Ancestors().OfType<TypeOfExpressionSyntax>().FirstOrDefault();
+            if (typeofExpression?.Type?.FullSpan.Contains(decl.Span) == true)
+            {
+                // invalid syntax case where operation is not supported
```

Comment:
>// invalid syntax case where operation is not supported [](start = 16, length = 55)

Please add a link to the issue tracking this bug.

Reply:
opened. https://github.com/dotnet/roslyn/issues/22175